### PR TITLE
fix(git-workspace-cleanup): detect stacked-PR merges via intermediate base branches

### PR DESCRIPTION
**Pull request by uda-lab-agent**
## Bug

`scripts/git_prune_worktrees.py` in `git-workspace-cleanup` fails to detect merged child PRs in stacked-PR workflows.

`gh_list_merged_prs` builds a `gh pr list` command that filters by `--base origin/main`. The 3-stage content check (exact OID / tree equality / ancestry) that follows operates on this already-filtered list. For stacked-PR workflows where children PRs merge into an intermediate integration branch (e.g., `mvp4`) and that integration branch is later merged into `main`, the children's PRs have `baseRefName = mvp4` (not `main`). The pre-filter returns `[]`, the content checks never run, and the children get classified as `unmerged` and skipped.

## Concrete failure (the case that surfaced this)

`uda-lab/yomotsusaka` MVP-4 used the stacked-PR pattern:

- Children PRs #80, #81, #83-#87 had `base=mvp4`.
- Integration PR #88 (`mvp4` -> `main`) was squash-merged as commit `84002e04`.
- After merge, running `git_prune_worktrees.py --yes` classified all 8 child branches/worktrees as `unmerged` because the gh query filtered by `--base main` returned 0 results before the tree-equality check could run.

## Fix (two layers)

**1. `gh_list_merged_prs`** — drop `--base` from the gh query and add `baseRefName,mergeCommit` to `--json`. Now returns all merged PRs for the head, regardless of base. The author/base context is reattached at the verification step.

**2. `pr_merged_via_gh`** — after a content sub-check passes against the PR head, verify reachability into `base_commit` via a new `_reaches_base()` helper. The helper handles three cases:

- **direct**: PR's `baseRefName` equals the cleanup base branch (classic case, unchanged behavior).
- **stacked (merge-commit)**: PR's `mergeCommit.oid` is an ancestor of `base_commit` — the integration branch was merge-committed into `--base`, so the child's merge commit is already reachable.
- **stacked (squash-of-integration)**: recursively look up the next merged PR with `head = current PR's baseRefName`; require the current PR's merge commit is reachable from that next PR's `headRefOid`, and that next PR itself reaches base. Depth-limited to 8 with cycle detection via a `visited` set. This is the squash-merge case where the integration branch's history isn't preserved in `--base`, so the child's `mergeCommit.oid` is not an ancestor of `base_commit` directly, but the content is.

The call site at `process_repo()` is updated to pass `base_commit` through to `pr_merged_via_gh`.

## Safety invariant preserved

`git branch -D` still requires audited proof the content is in `--base`. The content checks (exact OID / tree equality / ancestry) still apply against the PR head; the new reachability check is an additional requirement, not a relaxation.

## Tested on

uda-lab/yomotsusaka MVP-4 case. Patched script's `--dry-run` queues 14 actions (7 worktrees + 7 branches), all with `reason: merged_*_via_pr` and the correct PR# detail. Before the patch, the same invocation returned 0 actionable items and 8 `unmerged` skips.

## Notes for reviewer

- The recursive `_reaches_base` helper handles BOTH merge-commit-stacked AND squash-of-integration scenarios — the squash case is the one that actually matters in practice for the MVP-4 squash workflow, but the merge-commit path is a cheap and correct fast path that subsumes the simpler shape.
- `_is_ancestor` falls back to fetching the OID from the remote if `git merge-base --is-ancestor` returns an unknown error, since the integration branch's commits may not be in the local clone after the integration branch was deleted post-merge.
- SKILL.md updated to document the new reachability sub-cases.

### CODE REVIEW

**Comment by t-uda** [comment]
auggie review

File: skills/git-workspace-cleanup/scripts/git_prune_worktrees.py [hunk 1]
```diff
@@ -488,17 +488,20 @@ def gh_list_merged_prs(
...
+    # base. Filtering by base here would drop those candidates before the
+    # content checks can run. Reachability into base_commit is re-verified in
+    # pr_merged_via_gh via the returned baseRefName / mergeCommit fields.
     command = [
         "gh", "pr", "list",
         "--repo", f"{owner}/{name}",
         "--state", "merged",
-        "--base", base,
-        "--json", "number,headRefName,headRefOid",
+        "--json", "number,headRefName,headRefOid,baseRefName,mergeCommit",
```
[the remark this answers is not in the document]
Comment (t-uda) [reply skills/git-workspace-cleanup/scripts/git_prune_worktrees.py:504]: Fixed in ee58545. The test harness in `install_fake_gh` now keys responses on head/search only (the `base` component of the caller-supplied tuple is retained for readability but is no longer part of the lookup key). The fake-gh binary no longer parses or uses `--base`. All fake PR payloads that represent direct merges have been updated to include `baseRefName` and `mergeCommit` fields so the in-process `_reaches_base()` check works correctly. A companion fix also handles the slash-base edge case (`release/foo` was being misparse as remote=`release`, branch=`foo`): `_reaches_base` now compares `baseRefName` against both the raw base string and the `base_branch_name`-stripped form. All 28 tests pass.

File: skills/git-workspace-cleanup/scripts/git_prune_worktrees.py [hunk 2]
```diff
@@ -566,7 +573,64 @@ def pr_merged_via_gh(
...
+        # (i.e., the integration branch contained our merge commit at the
+        # time it was merged onward), and that PR must itself reach base.
+        base_ref = item.get("baseRefName")
+        if not base_ref:
+            return False
+        visited = set() if visited is None else visited
+        if base_ref in visited:
+            return False
+        visited = visited | {base_ref}
+        next_items, _next_err = gh_list_merged_prs(repo, owner, name, head=base_ref)
```
[the remark this answers is not in the document]
Comment (t-uda) [reply skills/git-workspace-cleanup/scripts/git_prune_worktrees.py]: Fixed in ee58545. `_next_err` is no longer discarded. The recursive call site now stores the error in `_reachability_errors` (a mutable list in the enclosing `pr_merged_via_gh` scope) and immediately returns `False` so the recursion terminates cleanly. After both candidate loops complete, if `_reachability_errors` is non-empty and no qualifying PR was found, the first error is returned as the function's error result — consistent with how the top-level `gh_list_merged_prs` failure surfaces `pr_check_failed` to the caller.

File: skills/git-workspace-cleanup/SKILL.md [hunk 3]
```diff
@@ -103,6 +104,15 @@ A branch is considered merged when either condition holds:
      ancestor. Covers cases where the remote branch was extended beyond the
      local branch before merging.
 
+   Reachability into `--base` is verified by one of:
```
[the remark this answers is not in the document]
Comment (t-uda) [reply skills/git-workspace-cleanup/SKILL.md:107]: Fixed in ee58545. Added a third reachability bullet to SKILL.md §2 documenting the recursive squash-of-integration path: when neither direct-merge nor mergeCommit-ancestor holds, the code looks up a merged PR with `head = current PR's baseRefName`, checks that our `mergeCommit.oid` is an ancestor of that next PR's `headRefOid`, and then recursively verifies that next PR itself reaches `--base`. The description includes the depth cap (8) and cycle detection (visited set of branch names).
